tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,420 @@
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_abnormal_train_type(self):
+        """Raising ValueError with wrong train_type."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_abnormal_train_type")
+        train_type = "unexpected"
+        inputs = {
+            "task_type": "classification",
+            "train_type": train_type,
+            "workspace_path": workspace_path,
+            "otx_root": self.otx_root,
+        }
+        with pytest.raises(ValueError):
+            self.otx_builder.build_task_config(**inputs)
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_normal_model_type(self):
+        """Build workspace with model_type argments."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_normal_model_type")
+        model_type = "yolox"
+        inputs = {
+            "task_type": "detection",
+            "model_type": model_type,
+            "workspace_path": workspace_path,
+            "otx_root": self.otx_root,
+        }
+        self.otx_builder.build_task_config(**inputs)
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        assert template.name.lower() == model_type
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_abnormal_model_type(self):
+        """Raise ValueError when build workspace with wrong model_type argments."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_abnormal_model_type")
+        inputs = {
+            "task_type": "detection",
+            "model_type": "unexpected",
+            "workspace_path": workspace_path,
+            "otx_root": self.otx_root,
+        }
+        with pytest.raises(ValueError):
+            self.otx_builder.build_task_config(**inputs)
+
+    @e2e_pytest_unit
+    @pytest.mark.parametrize("backbone_type", ["mmcls.MMOVBackbone"])
+    def test_builder_build_backbone_config_generate_backbone(self, backbone_type):
+        """Generate backbone config file (mmcls.MMOVBackbone)."""
+        tmp_backbone_path = os.path.join(self.tmp_dir_path, "backbone.yaml")
+        self.otx_builder.build_backbone_config(backbone_type, tmp_backbone_path)
+        assert os.path.exists(tmp_backbone_path)
+        backbone_config = mmcv.load(tmp_backbone_path)
+        assert backbone_config["backbone"]["type"] == backbone_type
+
+    @e2e_pytest_unit
+    @pytest.mark.parametrize("backbone_type", ["mmcls.MMOVBackbone"])
+    def test_builder_build_backbone_config_abnormal_output_path(self, backbone_type):
+        """Raise ValueError with wrong output_path."""
+        tmp_backbone_path = os.path.join(self.tmp_dir_path, "wrong.path")
+        with pytest.raises(ValueError):
+            self.otx_builder.build_backbone_config(backbone_type, tmp_backbone_path)
```

Comment:
If I'm missing anything, these lines are not covered by the test cases. This would be an important test case. Would it be possible to write a test case for this one as well?

Reply:
I'm sorry but I didn't understand this comment. Are you saying it would be nice to add some more test cases related to this? (wrong output path test)